apollo-federation/src/operation/mod.rs

```diff
@@ -1174,10 +1325,29 @@ mod field_selection {
         }
     }
 
+    impl PartialEq for Field {
+        fn eq(&self, other: &Self) -> bool {
+            self.data.field_position.field_name() == other.data.field_position.field_name()
+                && self.key == other.key
+                && self.sorted_arguments == other.sorted_arguments
+        }
+    }
+
+    impl Eq for Field {}
+
+    impl Hash for Field {
```

Comment:
Not for now, but it would be interesting to use apollo-compiler's `Node<>` type instead of `Arc<>` in select places in this file, as it caches the hash.

Reply:
@goto-bus-stop 
Yeah, it would be nice to have automatic cache-hashing, it's certainly something we should consider in the future.